pkg/util/http/transport.go

```diff
@@ -0,0 +1,102 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package http
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// CreateHTTPTransport creates an *http.Transport for use in the agent
+func CreateHTTPTransport() *http.Transport {
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: config.Datadog.GetBool("skip_ssl_validation"),
+	}
+
+	if config.Datadog.GetBool("force_tls_12") {
+		tlsConfig.MinVersion = tls.VersionTLS12
+	}
+
+	// Most of the following timeouts are a copy of Golang http.DefaultTransport
```

Comment:
You could have instanciated it with the DefaultTransport and then specialized some values, this way you would always be up-to-date with defaults from the Go stdlib. That's a nit, feel free to ignore this is completely fine this way as well.

```
transport := http.DefaultTransport
transport.MaxIdleConns: XXX // specialization
```

Reply:
good point, but this is existing code that I left unchanged. We can follow up later to make this better.

Reply:
I wouldn't recommend using the global `DefaultTransport` var. Odd things will happen if somewhere somehow someone reassigns it. Can end up very hard to debug.

Reply:
True!